### PR TITLE
Rename shear rate variable in Bingham

### PR DESCRIPTION
**Pull request by ezrayst**
Since the apparent viscosity is divided by shear rate, we add a check to make sure it's not divided by zero.

### CODE REVIEW

File: include/material/bingham.tcc [hunk 1]
```diff
@@ -75,14 +75,17 @@ Eigen::Matrix<double, 6, 1> mpm::Bingham<Tdim>::compute_stress(
...
-  const double shear_rate_squared =
-      2 * (strain_rate.dot(strain_rate) +
-           strain_rate.tail(3).dot(strain_rate.tail(3)));
+  double shear_rate =
+      std::sqrt(2 * (strain_rate.dot(strain_rate) +
+                     strain_rate.tail(3).dot(strain_rate.tail(3))));
 
   // Apparent_viscosity maps shear rate to shear stress
+  // Check if shear rate is 0
   double apparent_viscosity = 0;
```
Comment (cbgeo) [inline include/material/bingham.tcc]: `0.`;

File: include/material/bingham.tcc [hunk 2]
```diff
@@ -75,14 +75,17 @@ Eigen::Matrix<double, 6, 1> mpm::Bingham<Tdim>::compute_stress(
...
-  if (shear_rate_squared > critical_shear_rate_ * critical_shear_rate_)
-    apparent_viscosity = 2 * ((tau0_ / (std::sqrt(shear_rate_squared))) + mu_);
+  if (shear_rate * shear_rate > critical_shear_rate_ * critical_shear_rate_) {
+    if (shear_rate < shear_rate_threshold) shear_rate = shear_rate_threshold;
+    apparent_viscosity = 2 * ((tau0_ / shear_rate) + mu_);
```
Comment (cbgeo) [inline include/material/bingham.tcc]: `2.`

**Review by cbgeo — CHANGES_REQUESTED** [review]

**Review by kks32 — APPROVED** [review]

**Review by cbgeo — APPROVED** [review]

**Comment by ezrayst** [comment]
I am sorry, accidentally overwrite your change, @kks32 . I have put them back

File: include/material/bingham.tcc [hunk 3]
```diff
@@ -75,14 +75,17 @@ Eigen::Matrix<double, 6, 1> mpm::Bingham<Tdim>::compute_stress(
...
+                     strain_rate.tail(3).dot(strain_rate.tail(3))));
 
   // Apparent_viscosity maps shear rate to shear stress
-  double apparent_viscosity = 0;
-  if (shear_rate_squared > critical_shear_rate_ * critical_shear_rate_)
-    apparent_viscosity = 2 * ((tau0_ / (std::sqrt(shear_rate_squared))) + mu_);
+  // Check if shear rate is 0
+  double apparent_viscosity = 0.;
+  if (shear_rate * shear_rate > critical_shear_rate_ * critical_shear_rate_) {
+    if (shear_rate < shear_rate_threshold) shear_rate = shear_rate_threshold;
```
Comment (kks32) [inline include/material/bingham.tcc]: I'm not convinced with this check. Is this when you set `critical_shear_rate_` to be zero? Even setting `shear_rate` to a small value like 1.e-15 would induce a very high viscosity!

**Review by kks32 — CHANGES_REQUESTED** [review]

File: include/material/bingham.tcc (hunk 3 above)
Comment (ezrayst) [reply include/material/bingham.tcc]: It's for both. `critical_shear_rate_` needs to be at certain value.
I know it's a high viscosity, but `shear_rate = 0` is even worse right?

Comment (kks32) [reply include/material/bingham.tcc]: `shear_rate` can't be zero inside the condition where `shear_rate^2` > critical_shear_rate^2. No reason to do the next check if it is for `0`

**Review by kks32 — CHANGES_REQUESTED** [review]

File: include/material/bingham.tcc (hunk 3 above)
Comment (ezrayst) [reply include/material/bingham.tcc]: Understood. I agree. I removed it but I would like to rename the variable to the one in this pull request

**Review by kks32 — APPROVED** [review]

**Pull request merged**